---
title: "Tutorial on `solarius` R package"
author: "Andrey Ziyatdinov, Helena Brunel"
date: "`r Sys.Date()`"
output:
  html_document:
    theme: united
    toc: true
  pdf_document:
    toc: true
bibliography: ref.bib
---

```{r}
opts_chunk$set(eval = TRUE, dev = "png")
```

## Abstract

In human genetics research, the study of complex disorders or related quantitative phenotypes 
is targeted to discover the underlying genetic architecture of such disorders.
Whereas simple Mendelian diseases are results of a mutation in a single gene,
for example [Huntington's disease](http://en.wikipedia.org/wiki/Huntington%27s_disease),
complex phenotypes are the product of the action of multiple genes and environmental factors
as well as their interactions.
For instance, obesity is a complex disease,
which genetic basis might be better dissected 
via measurable quantitative variation in phenotypes,
such as body-mass index, closely related to the disease risk.

One of the study designs to identify the quantitative trait loci (QTLs)
that affect complex phenotypes is based on examination 
of related individuals grouped by families (pedigrees).
Pedigree-based designs allow for both QTL linkage and association mappings.
In addition, QTL analysis with pedigrees 
possess more power to detect rare variants,
permits explicit control for population stratification in association studies,
and suit for examination of parent-of-origin and sex-specificity effects.

The analysis of the pedigree data is traditionally based on variance component (VC) model, 
also known as linear mixed model (LMM).
`SOLAR` is one of the well-known and widely used tools
that implements the variance component models and other routines needed for the QTL analysis.
Similar tools of the interest include `Mendel`, `MMM`, `GEMMA` and `FAST-LMM`.

This vignette introduces examples on fitting statistical models available in `SOLAR`.
Three types of models for quantitative genetic analysis in family-based studies are introduced:
polygenic, association and linkage.

## Introduction

### General mixed model

Consider a particular trait, such as body-mass index, 
is observed in $n$ individuals grouped by families or clusters of related individuals.
A $n \times 1$ vector $y$ contains the phenotypic values measured for the trait.
One assumes that these observations are described adequately by a linear model
with a $p \times 1$ vector of fixed effects ($\beta$)
and a $q \times 1$ vector of random effects ($u$) [@Lynch1998, Chapter 26, pp. 746].
In matrix form,

$$ y = X \beta + Z u + e $$

where $X$ and $Z$ are respectively $n \times p$ and $n \times q$ incidence matrices,
and $e$ is a $n \times 1$ vector of residual deviations
assumed to be distributed independently of the random effects.
Denote the $n \times $n covariance matrix for the vector $e$ by $R$
and the the $q \times q$ covariance matrix for the vector $u$ by $G$.

The first element of the vector $\beta$ is typically the population mean,
and other elements may be age, gender, diet and so on.
The elements of the incidence matrices are usually equal to 0 or 1,
that indicate the effect contribution to the individuals's phenotype.
The model is referred to as a __mixed model__,
because this model jointly accounts for fixed and random effects.

Giving the two terms of the random effects $u$ and $e$,
the first term typically accounts for the contribution from random genetic effects
such as additive genetic values.
The second term accounts for the residual variance,
and one usually assumes that residual errors have constant variance and are uncorrelated,
so that $R$ is a diagonal matrix, with $R = \sigma_e^2 I$.

Two more equations for the means and variances of the component vectors
of the mixed model can be derived on the basis of its definition.
Since $E(u) = E(e) = 0$ by definition,

$$ E(y) = X \beta $$

Excluding the difference among individuals due to fixed effects 
and calling the assumption that $u$ and $e$ are uncorrelated,
the covariance matrix for the vector of observations $y$ is

$$ Var(y) = V = Z G Z^{T} + R $$

Now the mixed model can be rewritten in a compact form

$$
y = X \beta + Z u + e, \mbox{ where } u \sim (0, G) \mbox{ and } e \sim (0, R) \\
\mbox{ implying } y \sim (X \beta, V) = (X \beta, Z G Z^T + R)
$$

For the mixed model,
$y$, $X$ and $Z$ are observed variables,
while $\beta$, $u$, $R$ and $G$ are generally unknown variables.
Thus, the analysis consists of two complementary estimation procedures:
(1) estimation of the covariance matrices $G$ and $R$,
and (2) estimation of the vectors of fixed and random effects, $\beta$ and $u$.
The covariance matrices are usually functions of a few unknown parameters 
or variance components.
For instance, the covariance matrix of the residuals errors
typically has the form $R = \sigma_e^2 I$,
where $\sigma_e^2$ is the parameter to be estimated.

The description of the two estimation procedures is out of scope of this document,
and the reader is referred to [@Lynch1998, Chapter 26-27], 
where the necessary theoretical material and accompanying examples are given.
On the other hand, the implementation of the estimation procedures in `SOLAR`
is discussed elsewhere [@Almasy1998].  

### Polygenic model in `SOLAR`

### Linkage model in `SOLAR`

### Association model in `SOLAR`

## Software

### Installation

### `solarius` R package 

```{r}
library(solarius)
```

### Data sets

* GAW10 data distributed within `SOLAR`
* Simulated data `dat30` from `multic` R package [@Andrade2006]
* Simulated data `dat50` from `FFBSKAT` R package
* GAIT1 data 

`dat50` data set is adapted from `example.data` data set in 
[FFBSKAT](http://mga.bionet.nsc.ru/soft/FFBSKAT/) R package. 
The following variables are in the data set.

* `phenodata`: A data.frame with the phenotypes `trait`, `sex` and `age`.
* `genodata`: A matrix of 50 genetic variants (columns) given for 66 individuals (rows).
  The genotypes are coded in the format as `1/1`, `1/2` and `2/2`.
* `genocovdata`: A matrix of covariates derived from the genotype data (additive model).
  The covariates are coded as integers 0, 1 and 2.
* `snpdata`: A data.frame of anotation for genetic variantes. 
   The variables iclude `name` of the variant, the chromose `chrom`, `position` in bp and `gene`.
* `kin`: A kinship matrix for 66 individuals.     


## Examples

### Loading data

### Data exploration

### Polygenic model (univariate)

```{r, cache = TRUE}
M1 <- solarPolygenic(trait1 ~ age + sex, dat30)
M1
```

#### Testing covariates

```{r, cache = TRUE}
M2 <- solarPolygenic(trait1 ~ age + sex, dat30, covtest = TRUE)
M2
```

```{r, cache = TRUE}
M2$cf
```

#### Custom kinship matrix

`dat50` data set is used.

```{r, cache = TRUE}
M3 <- solarPolygenic(trait ~ 1, phenodata, kinship = kin)
M3
```

The polygenic and sporadic models have the same LRT statistics.

```{r, cache = TRUE}
M3$lf
```

### Polygenic model (bivariate)

```{r, cache = TRUE}
B1 <- solarPolygenic(trait1 + trait2 ~ 1, dat30)
B1
```

#### Testing correlations

```{r, cache = TRUE}
B2 <- solarPolygenic(trait1 + trait2 ~ 1, dat30, polygenic.options = "-testrhoe -testrhog")
B2
```

```{r, cache = TRUE}
B2$vcf
```

#### Trait-specific covariates

```{r, cache = TRUE}
B3 <- solarPolygenic(trait1 + trait2 ~ sex + age(trait2), dat30)
B3
```

```{r}
l_ply(tail(B3$solar$files$model$poly.mod, 3), function(x) cat(x, "\n"))

```

### Association model

`dat50` data set is used.

#### SNP data by `snpdata` argument

```{r genodata, cache = TRUE}
A1 <- solarAssoc(trait ~ 1, phenodata, snpdata = genodata, kinship = kin)
A1
```

#### SNP data by `snpcovdata` argument

```{r genocovdata, cache = TRUE}
A2 <- solarAssoc(trait ~ 1, phenodata, snpcovdata = genocovdata, kinship = kin)
A2
```


```{r genocovdata2, cache = TRUE}
genocovdata2 <- genocovdata
genocovdata2[genocovdata2 == 2] <- 1.9

A2 <- solarAssoc(trait ~ 1, phenodata, snpcovdata = genocovdata2, kinship = kin)
A2
```


#### SNP data by `genocov.files` (single value)
 
```{r genocov, cache = TRUE}
dir <- system.file("extdata/solarAssoc/", package = "solarius")
genocov.files <- file.path(dir, "snp.genocov")
snplists.files <- file.path(dir, c("snp.geno-list1", "snp.geno-list2"))

A3 <- solarAssoc(trait ~ 1, phenodata, 
  genocov.files = genocov.files, snplists.files = snplists.files)
A3
```

#### SNP data by `genocov.files` (many values)
 
```{r genocov2, cache = TRUE}
dir <- system.file("extdata/solarAssoc/", package = "solarius")
genocov.files <- file.path(dir, c("snp.genocov1", "snp.genocov2"))
snplists.files <- file.path(dir, c("snp.geno-list1", "snp.geno-list2"))

A4 <- solarAssoc(trait ~ 1, phenodata, 
  genocov.files = genocov.files, snplists.files = snplists.files)
A4
```

#### Map information and Manhetten plot

```{r assoc_map, cache = TRUE}
A5 <- solarAssoc(trait ~ 1, phenodata, snpdata = genodata, snpmap = snpdata, kinship = kin)
```

```{r manh_plot}
plot(A5)
```

#### Parallelization

```{r assoc_par, cache = TRUE}
A6 <- solarAssoc(trait ~ 1, phenodata, snpdata = genodata, kinship = kin, cores = 2)
```

```{r}
A1$assoc$tprofile$cputime.sec
A6$assoc$tprofile$cputime.sec
```

### Linkage model

### Advanced models

## Miscellaneous material

### `SOLAR` references


A list of references given on the official web page of `SOLAR` 
[http://solar.txbiomedgenetics.org/](http://solar.txbiomedgenetics.org/):

* The main reference for SOLAR, including theoretical explanations of the variance component linkage method and the approximate multipoint IBD calculations in pedigrees, is [@Almasy1998];
* Bivariate Quantitative trait linkage is described in the following papers: [@Almasy1997], [@Williams1999];
* The Liability Threshold model for discrete traits is described in the preceding paper as well as the following one: [@Duggirala1997];
* Gene By Environment Interaction is discussed in: [@Towne1997], [@Blangero2009];
* An examination of LOD adjustment is given in: [@Blangero2000], [@Blangero2001];
* Additional references include: [@Blangero1997], [@Williams2004], [@Williams1999b].


## Refences
