---
title: "SOLAR models for quantitative genetic analysis"
author: "Andrey Ziyatdinov, Helena Brunel"
date: "`r Sys.Date()`"
output:
  html_document:
    theme: united
    toc: true
  pdf_document:
    toc: true
bibliography: ref.bib
---

```{r}
opts_chunk$set(eval = FALSE)
```

## About

This vignette introduces examples on fitting statistical models available in `SOLAR`.
Three types of models for quantitative genetic analysis in family-based studies are introduced:
polygenic, association and linkage.


## `SOLAR`

via [http://solar.txbiomedgenetics.org/](http://solar.txbiomedgenetics.org/)

The main reference for SOLAR, including theoretical explanations of the variance component linkage method and the approximate multipoint IBD calculations in pedigrees, is: 

* Almasy L, Blangero J (1998) Multipoint quantitative trait linkage analysis in general pedigrees. Am J Hum Genet 62:1198-1211.
Bivariate Quantitative trait linkage is described in the following papers:
* Almasy L, Dyer TD, Blangero J (1997) Bivariate quantitative trait linkage analysis: Pleiotropy versus co-incident linkages. Genet Epidemiol 14:953-958.
* Williams JT, Van Eerdewegh P, Almasy L, Blangero J (1999) Joint Multipoint Linkage Analysis of Multivariate Qualitative and Quantitative Traits. I. Likelihood Formulation and Simulation Results. Am J Hum Genet 65:1134-1147

The Liability Threshold model for discrete traits is described in the preceding paper as well as the following one:

* Duggirala R, Williams JT, Williams-Blangero S, and Blangero J (1997) A variance component approach to dichotomous trait linkage analysis using a threshold model. Genet Epidemiol 14:987-992.

Gene By Environment Interaction is discussed in:

* Towne B, Siervogel RM, Blangero J (1997) Effects of genotype-by-sex interaction on quantitative trait linkage analysis. Genet Epidemiol 14:1053-1058.
* Blangero J (1993) Statistical Genetic Approaches to Human Adaptability. Hum Biol 65:941-966.

An examination of LOD adjustment is given in:

* Blangero J, Williams JT, Almasy L (2000) Robust LOD scores for variance component-based linkage analysis. Genet Epidemiol 19 Suppl 1:S8-14.
* Blangero J, Williams JT, Almasy L (2001) Variance component methods for detecting complex trait loci. Adv Genet 42:151-81.

Additional references include:

* Blangero J, Almasy L (1997) Multipoint oligogenic linkage analysis of quantitative traits. Genet Epidemiol 14:959-964.
* Mitchell BD, Ghosh S, Schneider JL, Birznieks G, Blangero J (1997) Power of variance component linkage analysis to detect epistasis. Genet Epidemiol 14:1017-1022.
* Williams JT, Blangero J (1999) Power of variance component linkage analysis to detect quantitative trait loci. Ann Hum Genet 63(6):545-563.
* Williams JT, Blangero J (1999) Comparison of variance components and sibpair-based approaches to quantitative trait linkage analysis in unselected samples. Genet Epidemiol 16(2):113-134.

## Statistical models

\[
y_{n \times 1} = X \beta + a_{n \times 1} + e_{n \times 1}
\]

## Data sets

* GAW10 data distributed within `SOLAR`
* Simulated data `dat30` from `multic` R package [@Andrade2006]
* Simulated data `dat50` from `?` R package
* GAIT1 data 

## Structure of `solarius` R package 

```{r}
library(solarius)
```

## Polygenic model (univariate)

```{r, cache = TRUE}
M1 <- solarPolygenic(trait1 ~ age + sex, dat30)
M1
```

### Testing covariates

```{r, cache = TRUE}
M2 <- solarPolygenic(trait1 ~ age + sex, dat30, covtest = TRUE)
M2
```

```{r, cache = TRUE}
M2$cf
```

### Custom kinship matrix

`dat50` data set is used.

```{r, cache = TRUE}
M3 <- solarPolygenic(trait ~ 1, phenodata, kinship = kin)
M3
```

The polygenic and sporadic models have the same LRT statistics.

```{r, cache = TRUE}
M3$lf
```

## Polygenic model (bivariate)

```{r, cache = TRUE}
B1 <- solarPolygenic(trait1 + trait2 ~ 1, dat30)
B1
```

### Testing correlations

```{r, cache = TRUE}
B2 <- solarPolygenic(trait1 + trait2 ~ 1, dat30, polygenic.options = "-testrhoe -testrhog")
B2
```

```{r, cache = TRUE}
B2$vcf
```

### Trait-specific covariates

```{r, cache = TRUE}
B3 <- solarPolygenic(trait1 + trait2 ~ sex + age(trait2), dat30)
B3
```

```{r}
l_ply(tail(B3$solar$files$model$poly.mod, 3), function(x) cat(x, "\n"))

```

## Association model

`dat50` data set is used.

### SNP data by `snpdata` argument

```{r, cache = TRUE}
A1 <- solarAssoc(trait ~ 1, phenodata, snpdata = genodata, kinship = kin)
A1
```

### SNP data by `snpcovdata` argument

```{r, cache = TRUE}
A2 <- solarAssoc(trait ~ 1, phenodata, snpcovdata = genocovdata, kinship = kin)
A2
```


### SNP data by `genocov.files` (single value)
 
```{r, cache = TRUE}
dir <- system.file("extdata/solarAssoc/", package = "solarius")
genocov.files <- file.path(dir, "snp.genocov")
snplists.files <- file.path(dir, c("snp.geno-list1", "snp.geno-list2"))

A3 <- solarAssoc(trait ~ 1, phenodata, 
  genocov.files = genocov.files, snplists.files = snplists.files)
A3
```

### SNP data by `genocov.files` (many values)
 
```{r, cache = TRUE}
dir <- system.file("extdata/solarAssoc/", package = "solarius")
genocov.files <- file.path(dir, c("snp.genocov1", "snp.genocov2"))
snplists.files <- file.path(dir, c("snp.geno-list1", "snp.geno-list2"))

A4 <- solarAssoc(trait ~ 1, phenodata, 
  genocov.files = genocov.files, snplists.files = snplists.files)
A4
```

### Parallelization

```{r, cache = TRUE}
A5 <- solarAssoc(trait ~ 1, phenodata, snpdata = genodata, kinship = kin, cores = 2)
```

```{r}
A1$assoc$tprofile$cputime.sec
A5$assoc$tprofile$cputime.sec
```

## Linkage model

## Advanced models

## Large-scale study: GAIT1 phenotypes

### BMI

### Throm

## Refences
