---
layout: tutorial
title: Association model
title_id: assoc
---

```{r}
args(solarAssoc)
```

| snpformat |  snpdata        | dat.gen        | genocov          |
|-----------|-----------------|----------------|------------------|
| "/"       | {A/A, A/T, T/T} | as is snpdata  | {0, 1, 2}        |
| "012"     | {0, 1, 2}       | {1/1, 1/2, 2/2}| {0, 1, 2}        |
| "0.1"     | [0; 2]          | skipped        | as is in snpdata |
{: class="table"}


### `SOLAR` commands and files


Association analysis provided by `solarius` package via `solarAssoc` function
is based on `mga` command of `SOLAR`.
The following code shows an exact rules (selected in `solarius`) 
to conduct an association analysis on 2 cores in parallel.

    > mod <- solarAssoc(trait ~ 1, phenodata, snpdata = genodata, kinship = kin, cores = 2)
    > mod$solar$assoc$cmd
    [[1]]
    [1] "load pheno dat.phe"                                             
    [2] "load model trait/null0.mod"                                     
    [3] "outdir assoc1"                                                  
    [4] "mga -files snp.genocov -snplists snp.geno-list1 -out assoc.out1"
    
    [[2]]
    [1] "load pheno dat.phe"                                             
    [2] "load model trait/null0.mod"                                     
    [3] "outdir assoc2"                                                  
    [4] "mga -files snp.genocov -snplists snp.geno-list2 -out assoc.out2"


A list of specific `SOLAR` files are used in the calculations,
as described in the following table.

| `SOLAR` file | `solarAssoc` argument | Comments | 
|--------------|-----------------------|----------|
| genocov.files | Yes (alternative to `snpdata`) | Data files with genotype-covariates (e.g. file per chromosome) |
| genolist.file | No | Meta file with SNP names (internal file derived from genocov.files) |
| snplist.files | Yes (alternative to `snplist`) | Meta files with SNP names to analize |
| out.dirs | No | Output directories for association models like `mga.mod` (directory per core) |
| out.files | No | Output files to store association results (file per core) | 
{: class="table"}

Two files `genocov.files` and `genolist.file` are output files
of a group of commands, that convert genotypes to covariates.
These three commands are `snp load dat.snp`, `snp covar -nohaplos`, `snp unload`.
See `snpdata2solar` function for more details.

Other three files `snplist.file`, `out.dirs` and `out.files` 
are custom files by `solarius` package 
to control association analysis (list of snps to be analyzed, parallel computation).

| `SOLAR` file | Default value | Parallel computation on 2 cores | 
|--|--|
| genocov.files | "snp.genocov" | |
| genolist.file | "snp.geno-list" | |
| snplist.file | "snp.geno-list" | "snp.geno-list1", "snp.geno-list2"|
| out.dirs | "assoc" | "assoc1", "assoc2"|
| out.files | "assoc.out" | "assoc.out1", "assoc.out2"|
{: class="table"}


### Example of association model

```{r}
data(dat50)

mod <- solarAssoc(trait ~ 1, phenodata, snpdata = genodata, kinship = kin)
head(mod$snpf)
```

